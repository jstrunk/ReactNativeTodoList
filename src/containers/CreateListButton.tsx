import React from 'react';
import { View } from 'react-native';
import { Icon } from 'react-native-elements';
import styles from '../styles';

interface Props {
  toggleCreateListModal: () => void;
}

const CreateListButton = (props: Props): JSX.Element => {
  return (
    <View style={styles.createListButtonView}>
      <Icon type='font-awesome-5' name='plus-circle' solid={true} onPress={props.toggleCreateListModal} size={50} color='lightgreen' />
    </View>
  );
}

export default CreateListButton;