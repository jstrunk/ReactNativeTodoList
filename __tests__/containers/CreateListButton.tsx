import 'react-native';
import React from 'react';
import CreateListButton from '../../src/containers/CreateListButton';
import { render, fireEvent } from '@testing-library/react-native';

describe('CreateListButton container', () => {
  let toggleCreateListModal: jest.Mock<any,any>;

  beforeEach(() => {
    toggleCreateListModal = jest.fn();
  });

  it('renders an CreateListButton', () => {
    const { toJSON } = render(
      <CreateListButton toggleCreateListModal={toggleCreateListModal} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call toggleCreateListModal', async () => {
    const { UNSAFE_getByProps } = render(
      <CreateListButton toggleCreateListModal={toggleCreateListModal} />
    );

    const element = UNSAFE_getByProps({name: 'plus-circle'});

    fireEvent.press(element);

    expect(toggleCreateListModal).toHaveBeenCalledTimes(1);
  });
});