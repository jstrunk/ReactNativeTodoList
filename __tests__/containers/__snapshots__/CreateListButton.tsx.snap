// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateListButton container renders an CreateListButton 1`] = `
<View
  style={
    Object {
      "bottom": 80,
      "position": "absolute",
      "right": 50,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "lightgreen",
                "fontSize": 50,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
